### ISSUE TITLE
Document the option to collapse cells while displaying the first line

### ISSUE THREAD

**Issue by danmcmahill** [jupyterlab/jupyterlab#16179]
### Problem
I'm trying out Jupyter Lab 4.1.6 to see if I want to migrate away from the classic notebook.  There is one key thing I'm missing. 
 When working with large notebooks in the legacy jupyter notebook, I found that not only code folding by code blocks but folding an entire cell based on the first line comment was very useful.  This capability was in the codefolding extension.  See https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/codefolding/readme.html for details and in particular this is the functionality I'm looking for:

![image](https://github.com/jupyterlab/jupyterlab/assets/22477431/d8a1fc59-1205-4bed-9fae-6e17d4e0ebcf)

### Proposed Solution

See if the functionality from https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/master/src/jupyter_contrib_nbextensions/nbextensions/codefolding/firstline-fold.js
can be added into JupyterLab since code folding is now part of Jupyter Lab.

**Comment by krassowski** [comment]
> I found that not only code folding by code blocks but folding an entire cell based on the first line comment was very useful

JupyterLab has cell folding like that since 4.0: https://github.com/jupyterlab/jupyterlab/pull/14141

![collapseing](https://github.com/jupyterlab/jupyterlab/assets/5832902/78d60540-0a79-48ca-9c0f-54fb4b9bbc84)

Did you try that already? Your issue seems to suggest that you was not able to find how to activate this feature nor about this feature but I am not sure. Is this just a documentation problem?

**Comment by danmcmahill** [comment]
> Did you try that already? Your issue seems to suggest that you was not able to find how to activate this feature nor about this feature but I am not sure. Is this just a documentation problem?

I had looked in the documentation and looked for the little arrow for folding and totally missed that I could do what you demonstrated.  So yes, I think it is just a documentation problem.  Thank you so much!

**Comment by krassowski** [comment]
Thanks for confirming! Reopening, because lacks in documentations are sill issues that need addressing.

**Comment by kiansahafi** [comment]
hello there,
i am new to this project and wanted to help out with the documentations.
would anybody be kind enough to guide me through accomplishing this?
thanks in advance.

**Comment by alexander-kuruvilla** [comment]
is this issue closed? seems like the documentation has been updated.

**Comment by krassowski** [comment]
> is this issue closed? seems like the documentation has been updated.

This issue is open (see the green circle in top left). I do not see a PR that would update the documentation, can you point to a specific place where you see it udpated?

**Comment by alexander-kuruvilla** [comment]
@krassowski you are right, my mistake. I am new to this and confused this with the closed PR for the notebook. I would like to help out with this documentation is nobody is currently working on it. 
I was thinking of mentioning the code folding behavior just below the 'collapse and expand' portion in the notebook documentation with some supporting images showing how it will look expanded and collapsed:
![image](https://github.com/user-attachments/assets/47e1965d-0f32-48e1-b69b-992ee4e2f516)

Please let me know your thoughts on this. Thanks.

**Comment by 1himan** [comment]
This issue should be closed by now if I'm right? The documentation is updated by @alexander-kuruvilla , right?

**Comment by ApplestarGY** [comment]
Although the "cell folding" feature does fold cells when displaying the first line, there's a key difference. With the old nbextensions, I could use the arrow keys to navigate through folded cells and run them with "shift+enter." However, in JupyterLab, when a cell is folded, I can no longer select it with the keyboard. Instead, I have to click the cell with the mouse, which causes it to expand immediately.

**Comment by L4R5m** [comment]
Another difference  (along with the regression mentioned by @ApplestarGY) is that with the old nbextension you could collapse the text while editing the cell by moving the cursor to the start of the 1st line of the cell and pressing <Left-Arrow>.
And if I recall correctly you could re-expand the cell in edit mode by pressing <Right-Arrow>.
I haven't used nbextensions in a while, so apologies if I'm misstating functionality.

In JupyterLab you have to precisely click the vertical 'blue bar' with the mouse... keyboard solutions are preferable.

**Comment by g-divya** [comment]
Updates on this issue? If the issue still exists, could this be assigned to me please?

**Comment by aaron-seq** [comment]
@krassowski I would like to work on this specific documentation if no ones working on it.

**Comment by krassowski** [comment]
Please do!

**Comment by aaron-seq** [comment]
@krassowski Based on my analysis of the current docs and PR #16586, I have a plan to complete this task.

The existing PR #16586 added basic documentation about cell folding with first-line visibility, but based on the comments from @ApplestarGY and @L4R5m, there are additional behavioral differences between the old nbextensions and JupyterLab that need to be documented.

My approach:

1. Review the current docs/source/user/notebook.rst to identify where cell folding documentation should be enhanced
2. Document the first-line visibility feature more comprehensively, including keyboard shortcuts and mouse interactions
3. Add technical details about the differences from the legacy codefolding extension (keyboard navigation behavior with folded cells, edit mode collapse/expand functionality)
4. Test the actual behavior locally to ensure documentation accuracy
5. Add appropriate images showing both collapsed and expanded states if needed
6. Follow the JupyterLab documentation style guide (second person, active tense, avoid trivializing words)
7. Run documentation tests using pytest before submitting
8. Create a PR with clear commit messages and references to this issue

I'll start by forking the repo, setting up the local environment, and testing the actual cell folding behavior to ensure the documentation reflects the current implementation accurately. Will keep you updated on progress.

**Comment by aaron-seq** [comment]
@krassowski I've created PR #18097 with the enhanced documentation for cell folding behavior.

The PR adds a dedicated section explaining first-line visibility, interaction methods (mouse and keyboard), and technical differences from the legacy codefolding extension. I addressed the concerns raised by @ApplestarGY and @L4R5m about keyboard navigation and edit mode behavior.

Key additions:
- Clear explanation of first-line visibility when cells are collapsed
- Mouse and keyboard interaction documentation
- Behavioral notes about cell expansion on selection
- Comparison with legacy nbextensions for users migrating from classic Notebook

The documentation follows JupyterLab's style guidelines and maintains consistency with the existing structure. Ready for review.

**Comment by krassowski** [comment]
>  I addressed the concerns raised by @ApplestarGY and @L4R5m about keyboard navigation and edit mode behavior.

I think that rather than documenting the differences we should possibly extract these out to new issues about adding options to align the behaviour with the legacy extension if that makes sense. I will let users who reported them open new issues.

**Comment by DilkhushYash** [comment]
Hi @krassowski, I would like to work on this issue.
If it is still available, could you please assign it to me? Thank you.

**Comment by MohammedSaad2701** [comment]
Hi, I would like to work on this issue. Could you please assign it to me?

**Comment by Rahasyapandey** [comment]
Hi, I’m a beginner contributor and I’d like to work on this issue. Can I take this up?

**Comment by Samarth40** [comment]
Hi, I explored this issue and will check whether JupyterLab currently supports collapsing cells with first-line visibility. If it exists, I’ll add documentation for it, otherwise I’ll document the limitation or workaround. Please let me know if this approach works.
